### PR TITLE
Add management command for programs skill

### PR DESCRIPTION
**Pull request by uzairr**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

**Review by saleem-latif — APPROVED** [review]
Looks Good 👍

File: taxonomy/utils.py
Comment (AliAdnanSohail) [inline taxonomy/utils.py:112]: Should we name this method as `refresh_product_skills` and then rather `course_or_programs` we can simple use `products` to keep it consistent with https://github.com/openedx/taxonomy-connector/pull/99/files#diff-c16bb57d10bc9185110cec11767320e1aa531630fae2c130e353f7379e032885R158

File: taxonomy/utils.py
Comment (AliAdnanSohail) [inline taxonomy/utils.py:56]: Suggestion: we can replace `is_programs` to `product_type` and then check if `product_type` is `Program` or any other. This way we can scale this up for multiple products as well

File: taxonomy/exceptions.py
Comment (DawoudSheraz) [inline taxonomy/exceptions.py:21]: nit: is not found

File: taxonomy/migrations/0022_merge_20220819_0834.py
Comment (DawoudSheraz) [inline taxonomy/migrations/0022_merge_20220819_0834.py:14]: nit: you can remove your migration and recreate to not have this merge migration.

**Review by DawoudSheraz — COMMENTED** [review]
A few nits. Very nice refactoring in taxonomy/utils.py

**Review by AliAdnanSohail — APPROVED** [review]
LGTM! Just wait for the build to be green before merge

File: taxonomy/management/commands/refresh_course_skills.py
Comment (DawoudSheraz) [inline taxonomy/management/commands/refresh_course_skills.py:31]: nit: class level vars should be all caps

File: taxonomy/tasks.py
Comment (DawoudSheraz) [inline taxonomy/tasks.py:26]: since this method is not called from programs in any way, the string 'Course' can be mentioned in argument directly instead of having a variable.

File: taxonomy/utils.py
Comment (DawoudSheraz) [inline taxonomy/utils.py:62]: nit: Please add a comment that this UUID is the value from Discovery and not program_uuid in ProgramSkills.

File: taxonomy/utils.py
Comment (DawoudSheraz) [inline taxonomy/utils.py:71]: nit: Product model and its identifier field

File: taxonomy/management/commands/refresh_program_skills.py
Comment (DawoudSheraz) [inline taxonomy/management/commands/refresh_program_skills.py:27]: Our underlying assumption here is that Programs have been published/active, correct?

File: tests/management/test_refresh_course_skills.py
Comment (DawoudSheraz) [inline tests/management/test_refresh_course_skills.py:202]: where is %s variable defined? Should this have a proper value here?

File: tests/management/test_refresh_program_skills.py
Comment (DawoudSheraz) [inline tests/management/test_refresh_program_skills.py:107]: Same question for %s, shouldn't these be strings here instead of actual %s.

**Review by DawoudSheraz — APPROVED** [review]
Left a few nit comments and some questions to address before merging.

File: taxonomy/tasks.py
Comment (uzairr) [reply taxonomy/tasks.py:26]: hmm, i think passing arguments via variables is more acceptable compared to hard-coding, but it may debatable!

File: taxonomy/management/commands/refresh_program_skills.py
Comment (uzairr) [reply taxonomy/management/commands/refresh_program_skills.py:27]: yes! it is

File: tests/management/test_refresh_course_skills.py
Comment (uzairr) [reply tests/management/test_refresh_course_skills.py:202]: its value is dependent on the `product_type` passed to `refresh_product_skill` method. The following line is refactored in this PR so that a genericness would be created.
https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/utils.py#L198

File: tests/management/test_refresh_program_skills.py
Comment (uzairr) [reply tests/management/test_refresh_program_skills.py:107]: same reason as above!

File: tests/management/test_refresh_course_skills.py
Comment (DawoudSheraz) [reply tests/management/test_refresh_course_skills.py:202]: Yes, that is true. But in the tests, the messages being compared should have a valid value against %s, not hardcoded %s.

File: taxonomy/tasks.py
Comment (DawoudSheraz) [reply taxonomy/tasks.py:26]: Only if the variable is used more than once. There is only one instance of usage here.

File: tests/management/test_refresh_course_skills.py
Comment (uzairr) [reply tests/management/test_refresh_course_skills.py:202]: i was of the same view while writing them but that's not how it works! I think it is resolved at the time of displaying them on screen otherwise they are treated as plain string 🤔 
e.g. https://github.com/openedx/taxonomy-connector/blob/master/tests/management/test_refresh_course_skills.py#L100

**Pull request merged**
